components/Feature/TopicExplorer/index.js

```diff
@@ -1,25 +1,39 @@
-import React, { useState } from 'react';
+import React, { useState, useEffect } from 'react';
 import Markdown from 'markdown-to-jsx';
 
 const TopicExplorer = (props) => {
   const [searchTerm, setSearchTerm] = useState('')
   const [searchResults, setSearchResults] = useState([])
+  const [tags, setTag] = useState([])
 
   const getSearchResults = (searchTerm) => {
     var results = []
     const newSearchTerm = searchTerm.toLocaleLowerCase();
-
     for(const topic of props.topics) {
       const promptTags = topic.promptTags
       promptTags.forEach(tag => {
-        if(tag.toLocaleLowerCase() == newSearchTerm){
-          results.push(topic);
-        }
+          if(tag.toLocaleLowerCase() == newSearchTerm){
+            results.push(topic);
+          }
       });
     }
     return results
   }
 
+  useEffect(() => {
+      if(props.topics){
+        props.topics.forEach(topic => {
+          topic.promptTags.forEach(promptTag => {
+            if(!tags.includes(promptTag)){
+              const newTags = tags
+              newTags.push(promptTag)
+              setTag(newTags)
+            }
+          });
+        });
+      }
```

Comment:
nit: I think these lines are indented one step too much?

```suggestion
    if(props.topics){
      props.topics.forEach(topic => {
        topic.promptTags.forEach(promptTag => {
          if(!tags.includes(promptTag)){
            const newTags = tags
            newTags.push(promptTag)
            setTag(newTags)
          }
        });
      });
    }
```